bundles/http/Citrix/00000019-0000-0000-0000-000000000000/47a6eb95-6f9d-44df-8be7-2322ada5770c.js

```diff
@@ -0,0 +1,864 @@
+const limit = 100 // limit number of records per page for pagination
+const maxNoOfPages = 100000 //  limit number of pages for pagination
+
+function calculateNumberOfPages(totalrecords) {
+	if (Number.isInteger(totalrecords)) {
+		let totalpages = totalrecords / limit
+		if (totalpages > maxNoOfPages) {
+			console.log(`Warning: Number of pages for pagination exceed set value ${maxNoOfPages}. Total number of pages needed: ${totalpages}`)
+			totalpages = maxNoOfPages
+		}
+		return totalpages
+	}
+	else {
+		console.log(`Warning: unable to calculate number of pages, total records returs as: ${totalrecords}`)
+		return 0
+	}
+}
+
+function flatComments(comments) {
+	for (let i = 0; i < comments.length; i++) {
+		//this will move key/value pairs from json.comments[j].user object to the json.comments[j] and add user_ prefix
+		comments[i] = flatJsonObject(comments[i], 'user', 'user_')
+	}
+	return comments
+}
+
+function flatLineItems(items) {
+	for (let i = 0; i < items.length; i++) {
+		items[i] = flatJsonObject(items[i], 'supplier', 'supp_')
+		items[i] = flatJsonObject(items[i], 'accountCategory', 'acc_cat_')
+		items[i] = flatJsonObject(items[i], 'description', 'desc_')
+		items[i] = flatJsonObject(items[i], 'desc_price', 'desc_pr_')
+	}
+	return items
+}
+
+function flatLineItemsAccountings(accountings) {
+	for (let i = 0; i < accountings.length; i++) {
+		accountings[i] = flatJsonObject(accountings[i], 'generalLedger', 'gl_')
+		accountings[i] = flatJsonObject(accountings[i], 'costCenter', 'cc_')
+		accountings[i] = flatJsonObject(accountings[i], 'amount', 'am_')
+	}
+	return accountings
+}
+
+function flatRequisition(requisition) {
+	requisition = flatJsonObject(requisition, 'totalCost', 'tc_')
+	requisition = flatJsonObject(requisition, 'requester', 'rq_')
+	return requisition
+}
```

Comment:
Are you sure for all of these `flatSomething` functions that input parameter is always array?
If not, then `array.length` will fail.
If you are not sure, then you can initialize input parameter with just
```js
function flatLineItemsAccountings(accounting = []) {
```